monai/apps/auto3dseg/hpo_gen.py

```diff
@@ -0,0 +1,263 @@
+        logger.info("Then add the following line to the trialCommand in your NNI config: ")
+        logger.info(f"{hint} {{remote_base_task_dir}} {{result_dir}} {{remote_algorithm_templates_dir}}")
+        logger.info("=" * 140)
+
+    def get_hyperparameters(self):
+        """
+        Get parameter for next round of training from nni server
+        """
+        if has_nni:
+            return nni.get_next_parameter()
+        else:
+            return {}
+
+    def update_params(self, params: dict):  # type: ignore
+        """
+        Translate the parameter from monai bundle to nni format.
+
+        Args:
+            params: a dict of parameters.
+        """
+        self.params = params
+
+    def get_task_id(self):
+        """
+        Get the identifier of the current experiment. In the format of listing the searching parameter name and values
+        connected by underscore in the file name.
+        """
+        task_id = ""
+        for k, v in self.params.items():
+            task_id += f"_{k}_{v}"
+        if len(task_id) == 0:
+            task_id = "_None"  # avoid rewriting the original
+        return task_id
+
+    def generate(self, output_folder: str = ".") -> None:
+        """
+        Generate the record for each Algo. If it is a BundleAlgo, it will generate the config files.
+
+        Args:
+            output_folder
+        """
+        task_id = self.get_task_id()
+        task_prefix = os.path.basename(self.algo.get_output_path())
+        write_path = os.path.join(output_folder, task_prefix + task_id)
+        self.obj_filename = os.path.join(write_path, "algo_object.pkl")
+
+        if isinstance(self.algo, BundleAlgo):
+            self.algo.export_to_disk(output_folder, task_prefix + task_id)
+            algo_export_to_pickle(self.algo, self.obj_filename, self.algo_templates_dir)
+        else:
+
+            ConfigParser.export_config_file(self.params, write_path)
+            logger.info(write_path)
+
+    def set_score(self, acc):
+        """
+        Report the acc to nni server
+        """
+        if has_nni:
+            nni.report_final_result(acc)
+        return
+
+    def run_algo(self, obj_filename: str, output_folder: str = ".", algo_templates_dir=None) -> None:  # type: ignore
+        """
+        The python interface for NNI to run
+
+        Args:
+            obj_filename: the pickle-exported Algo object.
+            output_folder: the root path of the algorithms templates.
+            algo_templates_dir: the algorithm_template. It must contain algo.py in the follow path:
+                ``{algorithm_templates_dir}/{network}/scripts/algo.py``
+        """
+        if not os.path.isfile(obj_filename):
+            raise ValueError(f"{obj_filename} is not found")
+
+        self.algo, self.algo_templates_dir = algo_import_from_pickle(obj_filename, algo_templates_dir)
+
+        # step1 sample hyperparams
+        params = self.get_hyperparameters()
+        # step 2 update model
```

Comment:
update algo?
```suggestion
        # step 2 update algo
```

Reply:
It should be "get the update params for the algo to run in the next trial". Thanks!